### PR TITLE
Java Interview Question set

### PR DESCRIPTION
**Pull request by 12rashic**
Added Java interview questions that  will be helpfull while learning Java  & that will also contribute to this repo.
image
Issue #3545

Kindly approve the PR & add for hacktoberfest label.
Thank You.

### CODE REVIEW

File: src/test/java/Java_interview_question/Java_interview_question.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+| -------------                                    | -------------                                                                                  |
+| 1.C++ is platform-dependent.                     | 1 .Java is platform-independent.                                                               |
+| 2.C++ supports multiple inheritance.             | 2.Java doesn't support multiple inheritance through class. Can be achieved by using Interface. |
+| 3.C++ supports operator overloading.             | 3.Java doesn't support operator overloading.                                                   |
+
+## 6.How many types of constructors are used in Java? 
+ **Default Constructor**: *default constructor is the one which does not accept any value. The default constructor is mainly used to initialize the instance variable with the default values.* 
+
+ **Parameterized Constructor** : *The parameterized constructor is the one which can initialize the instance variables with the given values*
+
```
Comment (Teuton-Roy) [inline src/test/java/Java_interview_question/Java_interview_question.md]: ### **_Example,_** For Better Understanding...

Code:-
 //Example of parametarized constructor//

> public class ParametarizedConstructor {
>     int id;
>     String name;
>     ParametarizedConstructor(int i, String n){
>         id = i;
>         name = n;
>     }
>     void display(){
>         System.out.println(id+" "+name);
>     }
>     public static void main(String[] args) {
>         ParametarizedConstructor p1 = new ParametarizedConstructor(1, "Teuton");
>         ParametarizedConstructor p2 = new ParametarizedConstructor(2, "Rashic");
>         p1.display();
>         p2.display();
>     }
> }
> 
> 
> 
>

File: src/test/java/Java_interview_question/Java_interview_question.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+## 5.Difference between C++ & Java ? 
+| C++                                              | Java                                                                                           |
+| -------------                                    | -------------                                                                                  |
+| 1.C++ is platform-dependent.                     | 1 .Java is platform-independent.                                                               |
+| 2.C++ supports multiple inheritance.             | 2.Java doesn't support multiple inheritance through class. Can be achieved by using Interface. |
+| 3.C++ supports operator overloading.             | 3.Java doesn't support operator overloading.                                                   |
+
+## 6.How many types of constructors are used in Java? 
+ **Default Constructor**: *default constructor is the one which does not accept any value. The default constructor is mainly used to initialize the instance variable with the default values.* 
+
```
Comment (Teuton-Roy) [inline src/test/java/Java_interview_question/Java_interview_question.md]: ### **_Example, For Better Understanding..._**
_Code:-_
//java default constuctor//
```

> public class DefaultConstructor {
>     int id;
>     String name;
>     void display()//Default constructor//
>     {
>         System.out.println(id+" "+ "Teuton" +" "+name);
>     }
>     public static void main(String[] args) {
>         DefaultConstructor d1 = new DefaultConstructor();
>         DefaultConstructor d2 = new DefaultConstructor();
>         d1.display();
>         d2.display();
>     }
> }
> 

```

**Review by Teuton-Roy — COMMENTED** [review]
Constructor example added

**Comment by siriak** [comment]
We don't add interview questions

**Pull request closed without merging**